pkg/virt-handler/cmd-client/client.go

```diff
@@ -233,9 +238,33 @@ func IsDisconnected(err error) bool {
 	return false
 }
 
-func (c *VirtLauncherClient) SyncVirtualMachine(vmi *v1.VirtualMachineInstance) error {
-	return c.genericSendVMICmd("SyncVMI", c.v1client.SyncVirtualMachine, vmi)
+func (c *VirtLauncherClient) SyncVirtualMachine(vmi *v1.VirtualMachineInstance, options *VirtualMachineOptions) error {
+
+	vmiJson, err := json.Marshal(vmi)
+	if err != nil {
+		return err
+	}
+
+	optionsJson, err := json.Marshal(options)
+	if err != nil {
+		return err
+	}
+
+	request := &cmdv1.VMIRequest{
+		Vmi: &cmdv1.VMI{
+			VmiJson: vmiJson,
+		},
+		Options: optionsJson,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	response, err := c.v1client.SyncVirtualMachine(ctx, request)
+
+	err = handleError(err, "SyncVMI", response)
+	return err
 
+	// return c.genericSendVMICmd("SyncVMI", c.v1client.SyncVirtualMachine, vmi)
```

Comment:
I feel it would be a overhead for `genericSendVMICmd` to expect `Options` because they're not not needed for any other operations except sync and maybe create in future.
Do you think it's a hard requirement to move it to `genericSendVMICmd`?

Reply:
your thoughts here @vladikr @rmohr ?

Reply:
I agree that it would be nice to extend `genericSendVMICmd`, but from my pov it's not a hard requirement at the moment. At some point, we could generalize `genericSendVMICmd` as `MigrateVirtualMachine` copies it too.

Reply:
Please remove the remarked function.